src/uri/ersip_uri.erl

```diff
@@ -694,7 +694,9 @@ assemble_headers(Headers) ->
 
 -spec raw_param({uri_param_name(), term()}) -> {binary(), binary()} | binary().
 raw_param({transport, Value}) -> {<<"transport">>, ersip_transport:assemble_bin(Value)};
-raw_param({maddr, Host})      -> {<<"maddr">>,    ersip_host:assemble_bin(Host)};
+raw_param({method, Value})
+  when is_binary(Value)       -> {<<"method">>, Value};
```

Comment:
Method everywhere in APIs is ersip_method:method(). So {method, ersip_method:method()} is more suitable here.

Reply:
this function used in `assemble_param` in `assemble_params` in `assemble_data` in `assemble` in `assemble_bin`, that expects `iolist` to pass it to `iolist_to_binary`, but it got list with atoms and crashes with `bad argument`

Reply:
Example (before fix):
```erlang
Raw = {sipmsg,{message,{response,609,<<>>},#{{hdr_key,allow}=>{header,<<65,108,108,111,119>>,{hdr_key,allow},[[<<66,89,69>>]]},{hdr_key,callid}=>{header,<<67,97,108,108,45,73,100>>,{hdr_key,callid},[<<40>>]},{hdr_key,content_type}=>{header,<<67,111,110,116,101,110,116,45,84,121,112,101>>,{hdr_key,content_type},[[[<<71>>,<<47>>,<<65>>,[]]]]},{hdr_key,cseq}=>{header,<<67,83,101,113>>,{hdr_key,cseq},[[<<48>>,<<32>>,<<79,80,84,73,79,78,83>>]]},{hdr_key,from}=>{header,<<70,114,111,109>>,{hdr_key,from},[[[<<67>>,<<32>>,60,[<<115,105,112>>,58,<<84,46,103>>],62],[[<<59>>,<<116,97,103>>,<<61>>,<<110>>]]]]},{hdr_key,maxforwards}=>{header,<<77,97,120,45,70,111,114,119,97,114,100,115>>,{hdr_key,maxforwards},[[<<49>>]]},{hdr_key,to}=>{header,<<84,111>>,{hdr_key,to},[[[<<109>>,<<32>>,60,[<<115,105,112>>,58,[[<<49>>,64],<<65,46,113>>,[58,<<54,48,51,57,54>>],[[<<59>>,<<108,114>>],[<<59>>,<<109,97,100,100,114>>,<<61>>,<<99,46,110>>],[<<59>>,method,<<61>>,<<97>>],[<<59>>,<<116,114,97,110,115,112,111,114,116>>,<<61>>,<<116,108,115>>],[<<59>>,<<116,116,108>>,<<61>>,<<51,56>>],[<<59>>,<<117,115,101,114>>,<<61>>,<<106>>]],[63,[<<98>>,61,<<77>>]]]],62],[[<<59>>,<<116,97,103>>,<<61>>,<<110>>]]]]},{hdr_key,via}=>{header,<<86,105,97>>,{hdr_key,via},[[<<83,73,80>>,47,<<50,46,48>>,47,<<87,83,83>>,<<32>>,<<65,46,77>>,[58,<<49,50,53,50,50>>],[]]]}},[],undefined},{method,<<79,80,84,73,79,78,83>>},undefined,#{allow=>{allow,{method_set,{1,{{method,<<66,89,69>>},nil,nil}}}},callid=>{callid,<<40>>},content_type=>{content_type,{mime,<<71>>,<<65>>},[]},cseq=>{cseq,{method,<<79,80,84,73,79,78,83>>},0},from=>{fromto,{display_name,<<67>>},{uri,{scheme,sip},{absolute_uri_data,<<84,46,103>>}},#{tag=>{tag,<<110>>}}},maxforwards=>{maxforwards,1},to=>{fromto,{display_name,[<<109>>]},{uri,{scheme,sip},{sip_uri_data,{user,<<49>>},{hostname,<<65,46,113>>},60396,#{lr=>true,maddr=>{hostname,<<99,46,110>>},method=><<97>>,transport=>{transport,tls},ttl=>38,user=><<106>>},#{<<98>>=><<77>>}}},#{tag=>{tag,<<110>>}}}},undefined},
ersip_sipmsg:assemble_bin(Raw)

% =>
An exception was raised: error:badarg.
Stacktrace: [{erlang,iolist_to_binary,
                     [[<<"SIP/2.0 ">>,<<"609">>,<<" ">>,<<>>,<<"\r\n">>,
                       <<"Via">>,<<": ">>,
                       [<<"SIP">>,47,<<"2.0">>,47,<<"WSS">>,<<" ">>,<<"A.M">>,
                        [58,<<"12522">>],
                        []],
                       <<"\r\n">>,<<"To">>,<<": ">>,
                       [[<<"m">>,<<" ">>,60,
                         [<<"sip">>,58,
                          [[<<"1">>,64],
                           <<"A.q">>,
                           [58,<<"60396">>],
                           [[<<";">>,<<"lr">>],
                            [<<";">>,<<"maddr">>,<<"=">>,<<"c.n">>],
                            [<<";">>,method,<<"=">>,<<"a">>],
                            [<<";">>,<<"transport">>,<<"=">>,<<"tls">>],
                            [<<";">>,<<"ttl">>,<<"=">>,<<"38">>],
                            [<<";">>,<<"user">>,<<"=">>,<<"j">>]],
                           [63,[<<"b">>,61,<<"M">>]]]],
                         62],
                        [[<<";">>,<<"tag">>,<<"=">>,<<"n">>]]],
                       <<"\r\n">>,<<"From">>,<<": ">>,
                       [[<<"C">>,<<" ">>,60,[<<"sip">>,58,<<"T.g">>],62],
                        [[<<";">>,<<"tag">>,<<"=">>,<<"n">>]]],
                       <<"\r\n">>,<<"Call-Id">>,<<": ">>,<<"(">>,<<"\r\n">>,
                       <<"CSeq">>,<<": ">>,
                       [<<"0">>,<<" ">>,<<"OPTIONS">>],
                       <<"\r\n">>,<<"Max-Forwards">>,<<": ">>,
                       [<<"1">>],
                       <<"\r\n">>,<<"Allow">>,<<": ">>,
                       [<<"BYE">>],
                       <<"\r\n">>,<<"Content-Type">>,<<": ">>,
                       [[<<"G">>,<<"/">>,<<"A">>,[]]],
                       <<"\r\nContent-Length: 0">>,<<"\r\n\r\n">>]],
                     []},
             {ersip_sipmsg,assemble_bin,1,
                           [{file,"/home/ne/Projects/ersip/_build/test/lib/ersip/src/message/ersip_sipmsg.erl"},
                            {line,334}]},
             {prop_basic,'-prop_is_assemble_parse_equal/0-fun-0-',1,
                         [{file,"/home/ne/Projects/ersip/test/prop_basic.erl"},
                          {line,44}]}].
```
problem with line ` [<<";">>,method,<<"=">>,<<"a">>]`  - method -- atom